course5/homework03/print.h

```diff
@@ -0,0 +1,11 @@
+#include <string>
+
+void print(int x);
+void print(long l);
+void print(bool b);
+void print(char x);
+void print(float f, int precision=0);
+void print(double d, int precision=0);
+void print(const char* str);
+void print(std::string s);
```

Comment:
Once you decided to do it the proper way, you could also wrap all the calls in a namespace.